yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -283,9 +291,38 @@ export class P2PClient implements P2P {
     this.p2pService.registerBlockReceivedCallback(handler);
   }
 
-  public requestTxByHash(txHash: TxHash): Promise<Tx | undefined> {
-    // Underlying I want to use the libp2p service to just have a request method where the subprotocol is defined here
-    return this.p2pService.sendRequest(TX_REQ_PROTOCOL, txHash);
+  /**
+   * Requests the transactions with the given hashes from the network.
+   *
+   * If a transaction can be retrieved, it will be returned, if not an undefined
+   * will be returned. In place.
+   *
+   * @param txHashes - The hashes of the transactions to request.
+   * @returns A promise that resolves to an array of transactions or undefined.
+   */
+  public requestTxs(txHashes: TxHash[]): Promise<(Tx | undefined)[]> {
+    const requestPromises = txHashes.map(txHash => this.requestTxByHash(txHash));
+    return Promise.all(requestPromises);
+  }
+
+  /**
+   * Uses the Request Response protocol to request a transaction from the network.
+   *
+   * If the underlying request response protocol fails, then we return undefined.
+   * If it succeeds then we add the transaction to our transaction pool and return.
+   *
+   * @param txHash - The hash of the transaction to request.
+   * @returns A promise that resolves to a transaction or undefined.
+   */
+  public async requestTxByHash(txHash: TxHash): Promise<Tx | undefined> {
+    const tx = await this.p2pService.sendRequest(TX_REQ_PROTOCOL, txHash);
```

Comment:
Should we have something all the way down here (or deeper) that just returns it if we already have it?

Reply:
Related to my second look comment. Would this not still be "happy" if we provide some other transaction than the one that actually matches `txHash`?

Reply:
The idea is that you would only call this function if you have already checked that you do not have it. I can include that in a comment to make sure this is the case.

I did not add explicit tests for this here, but I think that I should. We likely cannot rely on the underlying tx pool to perform this check